aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -0,0 +1,978 @@
+            CreateStakingContractEvent { operator, voter, pool_address, principal, commission_percentage },
+        );
+        pool_address
+    }
+
+    /// Add more stake to an existing staking contract.
+    public entry fun add_stake(staker: &signer, operator: address, amount: u64) acquires Store {
+        let staker_address = signer::address_of(staker);
+        assert_staking_contract_exists(staker_address, operator);
+
+        let store = borrow_global_mut<Store>(staker_address);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+
+        // Add the stake to the stake pool.
+        let staked_coins = coin::withdraw<AptosCoin>(staker, amount);
+        stake::add_stake_with_cap(&staking_contract.owner_cap, staked_coins);
+
+        staking_contract.principal = staking_contract.principal + amount;
+        let pool_address = staking_contract.pool_address;
+        emit_event(
+            &mut store.add_stake_events,
+            AddStakeEvent { operator, pool_address, amount },
+        );
+    }
+
+    /// Convenient function to allow the staker to update the voter address in a staking contract they made.
+    public entry fun update_voter(staker: &signer, operator: address, new_voter: address) acquires Store {
+        let staker_address = signer::address_of(staker);
+        assert_staking_contract_exists(staker_address, operator);
+
+        let store = borrow_global_mut<Store>(staker_address);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+        let pool_address = staking_contract.pool_address;
+        let old_voter = stake::get_delegated_voter(pool_address);
+        stake::set_delegated_voter_with_cap(&staking_contract.owner_cap, new_voter);
+
+        emit_event(
+            &mut store.update_voter_events,
+            UpdateVoterEvent { operator, pool_address, old_voter, new_voter },
+        );
+    }
+
+    /// Convenient function to allow the staker to reset their stake pool's lockup period to start now.
+    public entry fun reset_lockup(staker: &signer, operator: address) acquires Store {
+        let staker_address = signer::address_of(staker);
+        assert_staking_contract_exists(staker_address, operator);
+
+        let store = borrow_global_mut<Store>(staker_address);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+        let pool_address = staking_contract.pool_address;
+        stake::increase_lockup_with_cap(&staking_contract.owner_cap);
+
+        emit_event(&mut store.reset_lockup_events, ResetLockupEvent { operator, pool_address });
+    }
+
+    /// Unlock commission amount from the stake pool. Operator needs to wait for the amount to become withdrawable
+    /// at the end of the stake pool's lockup period before they can actually can withdraw_commission.
+    public entry fun request_commission(operator_account: &signer, staker: address) acquires Store {
```

Comment:
do we really want `operator_account` or is `operator` -- the address, sufficient?

Reply:
Operator might not want to withdraw any commission for an extended period of time to maximize compounding. If this function only requires operator address, anyone can call it and affect the operator's commission compounding.

Reply:
Updated to be callable by either staker or operator.